### PR TITLE
added 'remove_empty_units' flag to export_to_phy funciton

### PR DESCRIPTION
**Pull request by Shawn-Guo-CN**
add a 'remove_empty_units' flag to function **export_to_phy**, in order to remove empty units that may appear in input sortings.

### CODE REVIEW

**Comment by colehurwitz** [comment]
Looks good! One question, why would the user ever set this flag to false? Would that throw an error?

**Comment by Shawn-Guo-CN** [comment]
> Looks good! One question, why would the user ever set this flag to false? Would that throw an error?

Just in case that the user deeply believe that there shouldn't be any empty unit in their data. Of course, I'm not sure whether this assumption is meaningful in real application scenarios.

**Comment by colehurwitz** [comment]
I think export to phy should just automatically get rid of empty units. I don’t think we need a flag for it. I would start with that.

File: spiketoolkit/postprocessing/postprocessing_tools.py [hunk 1]
```diff
@@ -733,8 +733,18 @@ def export_to_phy(recording, sorting, output_folder, nPC=3, electrode_dimensions
...
+        _spikes = sorting.get_unit_spike_train(unit_id)
+        if _spikes.shape[0] == 0: _empty_flag = True
+    if _empty_flag:
+        print('Warning: empty units have been removed when being exported to Phy')
+        sorting = st.curation.threshold_num_spikes(sorting, 1)
+
     if len(sorting.get_unit_ids()) == 0:
-        raise Exception("No units in the sorting result, can't save to phy.")
+        raise Exception("No non-empty units in the sorting result, can't save to phy.")
+
```
Comment (samuelgarcia) [inline spiketoolkit/postprocessing/postprocessing_tools.py]: Here is another problem : no units at all. Please come back to the previsous Exception.

**Comment by samuelgarcia** [comment]
Hi.
thank you for this.
My idea was to add an option in the function signautre.
If you clean anyway, the first list is useless.
I think.

**Comment by Shawn-Guo-CN** [comment]
Now, what the function export_to_phy does is to check:

1. whether input recording and sorting has correct type;
2. whether there exists empty unit, remove them if so;
3. whether there are units in soring.

At the current time, it seems solid to export sortings to phy.

**Pull request closed without merging**